### DISCUSSION TITLE
How to assign the correct "assigned_object_type" to Journal

### DISCUSSION THREAD

**Discussion by nomad-cyanide** [netbox-community/netbox#15551] [Help Wanted!]
Hi

I need to create journals for various objects in Netbox, but I am in need of a good way to obtain the value for the required field "assigned_object_type" before I can create the journal. 
Initially I thought it was enough the assign an existing object for the journal I want to create, but you have to set the type manually as well. My hope was that the journal creation process, could get the type based on the object assigned, but no such luck, so how do I assign the correct value dynamically?

E.g. if a journal needs to be created on a virtual cluster, the type value should be "virtualization.cluster", for a VM it must be "virtualization.virtualmachine" or for a device it must be "dcim.device"

I am trying to create function that will check for a specific tag with a journal, in order to check how long ago that journal was created and I would very much like to not have to make a function for each journal type:

Below works for VM's, but I would like to use the function for other objects in Netbox as well, like clusters, devices and others.
```
def check_journal_age(nb_object,nb_tag):
    nb_journal = ""
    nb_journals = nb.extras.journal_entries.filter(assigned_object_id=nb_object.id)
    if not nb_journals:
        log.info("Creating journal to be able to track timestamp.(1)")
        nb.extras.journal_entries.create(assigned_object_type="virtualization.cluster",assigned_object_id=nb_object.id,comments="Not found in source data - timestamp tracking",tags=[ nb_tag.id ])
    else:
        for journal in nb_journals:
            journal_tags = journal.tags
            if nb_tag in journal_tags:
                nb_journal = journal

        if not nb_journal:
            log.info("Creating journal to be able to track timestamp.(2)")
            nb.extras.journal_entries.create(assigned_object_type="virtualization.cluster",assigned_object_id=nb_object.id,comments="Not found in source data - timestamp tracking",tags=[ nb_tag.id ]) 

        else:
            creation_date = nb_journal.created
            creation_datetime_object = datetime.strptime(creation_date, '%Y-%m-%dT%X.%f%z')
            current_time = datetime.astimezone(datetime.now())
            elapsed_time=(current_time-creation_datetime_object)
            elapsed_time_seconds=int(elapsed_time.total_seconds())
            if elapsed_time_seconds > 1468800:
                # Call delete function
                nb_delete_object(nb_object)
 
```
Netbox version: v3.6.9

**Comment by candlerb** [comment]
> My hope was that the journal creation process, could get the type based on the object assigned

If you were using the Django ORM you could.

But here it looks like you're using the REST API (I'm presuming nb = pynetbox), in which case it's just JSON.

What type of object are you passing in as "nb_object" ?

**Reply by nomad-cyanide** [reply]
You are correct, I'm using pynetbox, to connect from our automation-server and update Netbox.

The type of object is an object fetched from Netbox, like a Virtual Cluster or a DCIM Device. Others may pop up as the project develops.

**Reply by markkuleinio** [reply]
So is your question basically "how to detect the object type of an object fetched with `pynetbox`?

**Reply by markkuleinio** [reply]
Simple solution is to let the caller tell it: if the caller knows it's dealing with VMs, it can pass a "virtualization.virtualmachine" string as an "object_type" argument, and so on

**Reply by markkuleinio** [reply]
Or, you can try to deduce it from the object URL, taking care of the plurals (`*/ipam/ip-addresses/*` -> `ipam.ip_address` etc).

**Reply by nomad-cyanide** [reply]
Thanks. Deducing it from the URL and manipulating it to the correct string, seems a little risky, so I have simply done this for the types I need.
```
 if "/dcim/devices/" in nb_object.url:
        object_type = "dcim.device"
```
It could likely be done in a better, smoother way, but this works fine for me. Thank you.

**Comment by nomad-cyanide** [comment]
Closing discussion
